src/source/Ice/Tls.c

```diff
@@ -0,0 +1,267 @@
+/**
+ * Kinesis Video Tcp
+ */
+#define LOG_CLASS "Tls"
+#include "../Include_i.h"
+
+STATUS createTlsSession(TlsSessionCallbacks callbacks, PTlsSession *ppTlsSession)
```

Comment:
I would prefer this to be a pointer instead of an entire struct being passed as a copy. In case of the pointer, change the name to pCallbacks and validate it together with ppTlsSession not being NULL